auslib/log.py

```diff
@@ -26,42 +26,118 @@ def makeRecord(self, name, level, fn, lno, msg, args, exc_info, func=None, extra
                 # Request object, whose id is actually what we want.
                 # Without this we end up with the id of the proxy object, which
                 # is static for the life of the application.
+                # TODO: this doesn't seem to be 100% unique.
+                # Sometimes requests that happen around the same time will
+                # end up with the same value here. Possibly only when the
+                # query strings are the same, or srcip?
                 requestid = id(request._get_current_object())
             # RuntimeError will be raised if there's no active request.
             except RuntimeError:
                 pass
             extra['requestid'] = requestid
-        return Logger.makeRecord(self, name, level, fn, lno, msg, args, exc_info, func, extra)
-
-
-# Standard CEF levels at Mozilla. More details here:
-# https://mana.mozilla.org/wiki/display/SECURITY/CEF+Guidelines+for+Application+Development+at+Mozilla#CEFGuidelinesforApplicationDevelopmentatMozilla-apxaAppendixA-%C2%A0Severity%C2%A0IntegersSuggestions
-CEF_INFO = 4
-CEF_WARN = 6
-CEF_ALERT = 8
-CEF_EMERG = 10
-
-
-def cef_event(name, severity, **custom_exts):
-    # Extra values need to be in the format csNLabel=xxx, csN=yyy
-    extra_exts = {}
-    n = 2
-    for k, v in custom_exts.iteritems():
-        valueKey = 'cs%d' % n
-        labelKey = '%sLabel' % valueKey
-        extra_exts[labelKey] = k
-        extra_exts[valueKey] = v
-        n += 1
-
-    username = request.environ.get('REMOTE_USER', 'Unknown User')
-    cef.log_cef(name, severity, request.environ, cef_config, username=username, **extra_exts)
-
-
-def get_cef_config(logfile):
-    return {
-        'cef.file': logfile,
-        'cef.version': 0,  # This is the CEF format version
-        'cef.product': 'Balrog',
-        'cef.vendor': 'Mozilla',
-        'cef.device_version': '1.0',
+        return logging.Logger.makeRecord(self, name, level, fn, lno, msg, args, exc_info, func, extra)
+
+
+class JsonLogFormatter(logging.Formatter):
+    """Log formatter that outputs machine-readable json.
+
+    This log formatter outputs JSON format messages that are compatible with
+    Mozilla's standard heka-based log aggregation infrastructure.
+
+    See also:
+    https://mana.mozilla.org/wiki/display/CLOUDSERVICES/Logging+Standard
+    https://mana.mozilla.org/wiki/pages/viewpage.action?pageId=42895640
+
+    Adapted from:
+    https://github.com/mozilla-services/mozservices/blob/master/mozsvc/util.py#L106
```

Comment:
Lets change master to a specific revision, or otherwise note what revision you used.